Support Batch Document Get

MajdSehwail opened the pull request:
Note that this requires the project's URL in the config to be changed from `projects/xxx/databases/(default)/documents` to `projects/xxx/databases/(default)`

MajdSehwail committed:
Support batch document get method

mujahed-odeh commented on lib/firestore/repo.ex:
```diff
@@ -78,60 +84,91 @@ defmodule Firestore.Repo do
         end
       end
 
+      def batch_get_documents(document_paths, params \\ []) do
```
document_ids or document_tokens

MajdSehwail committed:
- Implement a decoder for batch get document struct instead of parsing it inside the repo method.
- Include the path for each document since the order of the response isn't guaranteed to match the input.
- Since we include each path for each requested document, it's better not to filter out missing documents. Include them with value `nil` for document.
- Fix the capability of passing body params

MajdSehwail committed:
Fix callback contract
